ppc64le: build failure on GCC <= 7 and clang <= 6

##### System information (version)
- OpenCV => after #15339
- Operating System / Platform => ppc64le GNU/Linux
- Compiler => GCC(4.9, 5, 6, 7), clang(4, 5, 6)

##### Detailed description
##### GCC ERROR (resovled by #15511)
````
In file included from /worker/buildbot/Power9_Linux_gcc-7__opencv/opencv/modules/core/src/matmul.dispatch.cpp:50:0:
/worker/buildbot/Power9_Linux_gcc-7__opencv/opencv/modules/core/src/matmul.simd.hpp: In function 'double cv::cpu_baseline::dotProd_32s(const int*, const int*, int)':
/worker/buildbot/Power9_Linux_gcc-7__opencv/opencv/modules/core/src/matmul.simd.hpp:2511:55: error: invalid parameter combination for AltiVec intrinsic __builtin_vec_mule
         a += v_cvt_f64(v_int64(vec_mule(s1.val, s2.val)));
                                                       ^
/worker/buildbot/Power9_Linux_gcc-7__opencv/opencv/modules/core/src/matmul.simd.hpp:2512:55: error: invalid parameter combination for AltiVec intrinsic __builtin_vec_mulo
         b += v_cvt_f64(v_int64(vec_mulo(s1.val, s2.val)));
                                                       ^
make[2]: *** [modules/core/CMakeFiles/opencv_core.dir/build.make:558: modules/core/CMakeFiles/opencv_core.dir/src/matmul.dispatch.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[  7%] Building CXX object 3rdparty/protobuf/CMakeFiles/libprotobuf.dir/src/google/protobuf/source_context.pb.cc.o
````

CI:
  - GCC 4.9  https://ocv-power.imavr.com/#/builders/5/builds/62
  - GCC 5 https://ocv-power.imavr.com/#/builders/16/builds/312
  - GCC 6 https://ocv-power.imavr.com/#/builders/1/builds/4615
  - GCC 7 https://ocv-power.imavr.com/#/builders/6/builds/53
  - GCC8 works fine

##### CLANG ERROR (should be resolved by  #15510 )
````
In file included from /worker/buildbot/Power8_Linux_clang-4__opencv/opencv/modules/core/src/matmul.dispatch.cpp:44:
In file included from /worker/buildbot/Power8_Linux_clang-4__opencv/opencv/modules/core/src/precomp.hpp:85:
In file included from /worker/buildbot/Power8_Linux_clang-4__opencv/opencv/modules/core/include/opencv2/core/hal/intrin.hpp:186:
/worker/buildbot/Power8_Linux_clang-4__opencv/opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp:1047:10: error: invalid operand in inline asm: 'xvcvsxddp ${0:x},${1:x}'
__asm__ ("xvcvsxddp %x0,%x1" : "=wa"(out) : "wa"(a.val));
         ^
/worker/buildbot/Power8_Linux_clang-4__opencv/opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp:1047:10: error: invalid operand in inline asm: 'xvcvsxddp ${0:x},${1:x}'
/worker/buildbot/Power8_Linux_clang-4__opencv/opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp:1047:10: error: unknown operand
<inline asm>:1:12: note: instantiated into assembly here
        xvcvsxddp ,
                  ^
In file included from /worker/buildbot/Power8_Linux_clang-4__opencv/opencv/modules/core/src/matmul.dispatch.cpp:44:
In file included from /worker/buildbot/Power8_Linux_clang-4__opencv/opencv/modules/core/src/precomp.hpp:85:
In file included from /worker/buildbot/Power8_Linux_clang-4__opencv/opencv/modules/core/include/opencv2/core/hal/intrin.hpp:186:
/worker/buildbot/Power8_Linux_clang-4__opencv/opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp:1047:10: error: invalid operand in inline asm: 'xvcvsxddp ${0:x},${1:x}'
__asm__ ("xvcvsxddp %x0,%x1" : "=wa"(out) : "wa"(a.val));
         ^
/worker/buildbot/Power8_Linux_clang-4__opencv/opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp:1047:10: error: invalid operand in inline asm: 'xvcvsxddp ${0:x},${1:x}'
/worker/buildbot/Power8_Linux_clang-4__opencv/opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp:1047:10: error: unknown operand
<inline asm>:1:12: note: instantiated into assembly here
        xvcvsxddp ,
````

CI:
 - CLANG 4 https://ocv-power.imavr.com/#/builders/3/builds/57
 - CLANG 5 https://ocv-power.imavr.com/#/builders/8/builds/53
 - CLANG 6 https://ocv-power.imavr.com/#/builders/15/builds/44  
 
##### Steps to reproduce

build OpenCV on ppc64le

--- a/modules/core/include/opencv2/core/vsx_utils.hpp
+++ b/modules/core/include/opencv2/core/vsx_utils.hpp
@@ -124,6 +124,33 @@ VSX_FINLINE(rt) fnm(const rg& a, const rg& b)  \
 
 #define VSX_IMPL_2VRG(rt, rg, opc, fnm) VSX_IMPL_2VRG_F(rt, rg, #opc" %0,%1,%2", fnm)
 
+#if __GNUG__ < 8
+
+    // Support for int4 -> dword2 expanding multiply was added in GCC 8.
+    #ifdef vec_mule
+        #undef vec_mule
+    #endif
+    #ifdef vec_mulo
+        #undef vec_mulo
+    #endif
+
+    VSX_REDIRECT_2RG(vec_ushort8,  vec_uchar16,  vec_mule, __builtin_vec_mule)
+    VSX_REDIRECT_2RG(vec_short8,  vec_char16,  vec_mule, __builtin_vec_mule)
+    VSX_REDIRECT_2RG(vec_int4,  vec_short8,  vec_mule, __builtin_vec_mule)
+    VSX_REDIRECT_2RG(vec_uint4,  vec_ushort8,  vec_mule, __builtin_vec_mule)
+    VSX_REDIRECT_2RG(vec_ushort8,  vec_uchar16,  vec_mulo, __builtin_vec_mulo)
+    VSX_REDIRECT_2RG(vec_short8,  vec_char16,  vec_mulo, __builtin_vec_mulo)
+    VSX_REDIRECT_2RG(vec_int4,  vec_short8,  vec_mulo, __builtin_vec_mulo)
+    VSX_REDIRECT_2RG(vec_uint4,  vec_ushort8,  vec_mulo, __builtin_vec_mulo)
+
+    // dword2 support arrived in ISA 2.07 and GCC 8+
+    VSX_IMPL_2VRG(vec_dword2,  vec_int4,  vmulesw, vec_mule)
+    VSX_IMPL_2VRG(vec_udword2, vec_uint4, vmuleuw, vec_mule)
+    VSX_IMPL_2VRG(vec_dword2,  vec_int4,  vmulosw, vec_mulo)
+    VSX_IMPL_2VRG(vec_udword2, vec_uint4, vmulouw, vec_mulo)
+
+#endif
+
 #if __GNUG__ < 7
 // up to GCC 6 vec_mul only supports precisions and llong
 #   ifdef vec_mul

